lib/elixir/lib/uri.ex

```diff
@@ -392,15 +397,15 @@ defmodule URI do
 
   def parse(%URI{} = uri), do: uri
 
-  def parse(s) when is_binary(s) do
+  def parse(string) when is_binary(string) do
     # From http://tools.ietf.org/html/rfc3986#appendix-B
     regex = ~r/^(([a-z][a-z0-9\+\-\.]*):)?(\/\/([^\/?#]*))?([^?#]*)(\?([^#]*))?(#(.*))?/i
-    parts = nillify(Regex.run(regex, s))
+    parts = nillify(Regex.run(regex, string))
 
     destructure [_, _, scheme, _, authority, path, _, query, _, fragment], parts
     {userinfo, host, port} = split_authority(authority)
 
-    scheme = normalize_scheme(scheme)
+    scheme = scheme && String.downcase(scheme)
```

Comment:
I thought it would be good to ditch `normalize_scheme/1` since we're processing other things (like the port) inline here, so slightly less noise :)